example/tests/example_test.py

```diff
@@ -1,22 +1,60 @@
 import unittest
 
-from snapstack import Runner
+from snapstack import Plan, Step, Setup, Cleanup
 
 
 class ExampleTest(unittest.TestCase):
 
-    def test(self):
-        r = Runner(
+    def test_example(self):
+        '''
+        _test_example_
+
+        Basic test, which installs and tests and example
+        snap, after letting the Plan object build the default
+        snapstack.
+
+        Note: this will install snapstack on your system!
+
+        '''
+        example = Step(
             snap='example',  # Name of the snap
-            location='./tests/',  # Parent location for tests and files
-                                  # below. {local} is a magic string that
-                                  # points at the 'tests' directory.
+            script_loc='./tests/',  # Parent location for tests and files
+                                    # below. {local} is a magic string that
+                                    # points at the 'tests' directory.
             tests=['example.sh'],  # Test the snap
             files=['example.json'],  # Config files
-            base=None  # Don't override default Base.
+            snap_store=False  # Build from local source, rather than
+                              # install from the snap store.
         )
+        plan = Plan(tests=[example])
+        plan()
+
+    def test_custom_base(self):
+        '''
+        _test_custom_base_
+
+        Reference code for adding and removing stuff from our Setup
+        and Cleanup.
+
+        '''
+        setup = Setup()
+        cleanup = Cleanup()
+
+        # Add a step.
+        example = Step(
+            snap='example',
+            script_loc='./tests/',
+            tests=['example.sh'],
+            files=['example.json'],
+        )
+        setup.add_steps(('example', example))
+
+        # Remove steps.
+        setup.remove_steps(
+            'keystone', 'nova', 'neutron', 'glance', 'nova_hypervisor',
+            'neutron_ext_net')
 
-        try:
-            r.run()
-        finally:
-            r.cleanup()
+        # Run with our changes (this will download the config files,
+        # but then install nothing but the example snap).
+        plan = Plan(tests=[], setup=setup.steps(), cleanup=cleanup.steps())
```

Comment:
@coreycb I've got a local change where I don't re-use the example script for a different purpose in the "custom base" example.

As far as cleanup goes ... I think the issue is probably that I have people passing in their custom tests to "tests", but bundling their custom cleanup in "cleanup". I like the fact that this gives us only three cleanly named variables, but I can see the argument for making things more consistent.